docs/config.md

```diff
@@ -74,23 +75,80 @@ Obviously, you would only set one line, don't copy them all in!
 
+## 3-way merge tools for conflict resolution
+
+The `ui.merge-editor` key specifies the tool used for three-way merge
+tools by `jj resolve`.  For example:
+
+    ui.merge-editor = "meld"  # Or "kdiff3" or "vimdiff"
+
+The "meld", "kdiff3", and "vimdiff" tools can be used out of the box,
+as long as they are installed.
+
+To use a different tool named `TOOL`, the arguments to pass to the tool
+MUST be specified in the `merge-tools.TOOL.merge-args` key. As an example
+of how to set this key and other tool configuration options, here is
+the out-of-the-box configuration of the three default tools. (There is
+no need to copy it to your config file verbatim, but you are welcome to
+customize it.)
+
+    # merge-tools.kdiff3.program  = "kdiff3"     # Defaults to the name of the tool if not specified
+    merge-tools.kdiff3.merge-args = ["$base", "$left", "$right", "-o", "$output", "--auto"]
+    merge-tools.meld.merge-args   = ["$left", "$base", "$right", "-o", "$output", "--auto-merge"]
+
+    merge-tools.vimdiff.merge-args = ["-f", "-d", "$output", "-M",
+                                      "$left", "$base", "$right",
+                                      "-c", "wincmd J", "-c", "set modifiable",
+                                      "-c", "set write"]
+    vimdiff.program = "vim"
+    merge-tools.vimdiff.merge-tool-edits-conflict-markers = true    # See below for an explanation
+
+`jj` replaces the following arguments with the appropriate file names:
+
+- `$output` (REQUIRED) is replaced with the name of the file that the
+merge tool will output. `jj` will read this file after the merge tool
+exits. The merge is considered to have failed if this file is empty,
+unchanged since `jj` wrote it, or if the merge tool exits with a non-zero
+exit code.  - `$left` and `$right` (REQUIRED) are replaced with the
+paths to two files containing the content of each side of the conflict.
+
+- `$base` (optional) is replaced with the path to a file containing the
+contents of the conflicted file in the last common ancestor of the two
+sides of the conflict.
+
+### Editing conflict markers with a tool
+By default, the merge tool starts with an empty output file. If the tool
+puts anything into the output file, and exits with the 0 exit code,
+`jj` assumes that the conflict is fully resolved. This is appropriate
+for most graphical merge tools.
+
+Tools like `vimdiff` are better suited to facilitate editing the conflict
+markers in `jj`'s default representation of the conflict. They set the
+`merge-tools.vimdiff.merge-tool-edits-conflict-markers = true` option. As
+a result, the output file starts out with this representation of the
+conflict. After the merge tool is done, if the output file still contains
+conflict markers, `jj` assumes that the conflict was only partially
+resolved and parses the conflict markers to get the new state of the
+conflict. The conflict is considered fully resolved when there are no
+conflict markers left.
```

Comment:
I think it's better to leave this out until we actually do that (we don't yet parse conflict markers, do we?)

Reply:
[We do](https://github.com/martinvonz/jj/pull/689/commits/036c0f7e3bfb487cec191a81fad9329e25f60e93#diff-f63a7a1f8fbd5317456532a629c49a1069755d404504f0fee76e17cf7655d009R273-R283). It is in that one commit that does way too many things.